hudi-cli/src/main/java/org/apache/hudi/cli/commands/CommitsCommand.java

```diff
@@ -431,4 +429,23 @@ public String syncCommits(@CliOption(key = {"path"}, help = "Path of the table t
     return "Load sync state between " + HoodieCLI.getTableMetaClient().getTableConfig().getTableName() + " and "
         + HoodieCLI.syncTableMetadata.getTableConfig().getTableName();
   }
+
+  /*
+  Checks whether a commit or replacecommit action exists in the timeline.
+  * */
+  private Option<HoodieCommitMetadata> getCommitMetadataForInstant(HoodieTimeline timeline, String instantTime) throws IOException {
+    List<HoodieInstant> instants = Arrays.asList(
+            new HoodieInstant(false, HoodieTimeline.COMMIT_ACTION, instantTime),
+            new HoodieInstant(false, HoodieTimeline.REPLACE_COMMIT_ACTION, instantTime),
+            new HoodieInstant(false, HoodieTimeline.DELTA_COMMIT_ACTION, instantTime));
+
+    Option<HoodieInstant> hoodieInstant = Option.fromJavaOptional(instants.stream().filter(timeline::containsInstant).findAny());
```

Comment:
minor: timeline.containsInstant is linear search in all active instants. Number of instants in timeline is expected to be small, so this may not be a big issue. if its not lot of work consider trim timeline to specified instant time using findInstantsBeforeOrEquals().getReverseOrderedInstants().findFirst()

Reply:
Ok, so instead of 3*n we do a linear search and then do 3 comparisons. Is this what you meant?
`Option<HoodieInstant> instant = Option.fromJavaOptional(timeline.findInstantsBeforeOrEquals(instantTime).getReverseOrderedInstants().findFirst());
    if (instant.isPresent()) {
      Option<HoodieInstant> hoodieInstant = Option.fromJavaOptional(instants.stream().filter(i -> i.equals(instant.get())).findAny());
      return hoodieInstant;
    }`